lib/utils/replace.go

```diff
@@ -58,9 +63,21 @@ func ReplaceRegexp(expression string, replaceWith string, input string) (string,
 	return ReplaceRegexpWith(expr, replaceWith, input)
 }
 
-// RegexpWithConfig compiles a regular expression given some configuration.
-// There are several important differences with standard lib (see ReplaceRegexp).
-func RegexpWithConfig(expression string, config RegexpConfig) (*regexp.Regexp, error) {
+type replaceKey struct {
+	expression string
+	ignoreCase bool
+}
+
+// replaceCache interns compiled regular expressions created in RegexpWithConfig
+// to improve performance.
+var replaceCache = mustCache[replaceKey, *regexp.Regexp](1000)
```

Comment:
Why a separate cache than `MatchString`? I'd rather see that turned into a strict LRU around `regexp.Compile` (i.e. using the very final string as the key) rather than hold two of those in memory, one of which is also carrying some extra flag as part of the key.

Reply:
If it's plausible that the same expression will be used in both contexts I think it's worth it to share the same cache.

I wasn't expecting `GlobToRegexp` to be this impactful, I wonder if it's the replacement of `\*` into `(.*)` through a regexp that's the issue - could you replace `GlobToRegexp` with `strings.ReplaceAll(regexp.QuoteMeta(in), ``\*``, ``(.*)``)` and see if the benchmark improves a bit?